xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultRefactoringConfiguration.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.internal;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.refactoring.internal.api.RefactoringConfiguration;
+
+/**
+ * Default implementation of {@link RefactoringConfiguration}.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+@Component
+@Singleton
+public class DefaultRefactoringConfiguration implements RefactoringConfiguration
+{
+    @Inject
+    @Named("xwikiproperties")
```

Comment:
No need to be that explicit IMO. The default ConfigurationSource is fine.

Reply:
Yes better to be able to change the config at all levels! Any reason you wanted to restrict that @manuelleduc ?

Reply:
Replaced with a dedicated configuration when moving the configuration to an XClass (https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-3c76751dca65c1a50961704e4a11dd04R43)